index.js

```diff
@@ -551,6 +551,150 @@ studio.internal = (function(proto) {
     }
   }
 
+  obj.SystemNode = function(studioURL, notificationListener) {
+    var studioURL = studioURL;
+    var notificationListener = notificationListener;
+    var appConnections = new Array();
+    var this_ = this;
+
+    this.onAppConnect = function(url, notificationListener) {
+      return new Promise(function (resolve, reject) {
+        var appConnection = new obj.AppConnection(url, notificationListener);
+        appConnections.push(appConnection);
+        var sys = appConnection.root();
+        sys.async.onDone(resolve, reject, sys);
+      });
+    };
+
+    this.onConnect = function(resolve, reject) {
+      this.onAppConnect(studioURL, notificationListener).then(function(system){
+        var promises = new Array();
+        system.forEachChild(function (app) {
+          if (!app.info().is_local)
+          {
+            var appUrl = app.info().server_addr + ":" + app.info().server_port;
+            promises.push(this_.onAppConnect(appUrl, notificationListener));
```

Comment:
I'm not sure I understand how it works. When apps are started and stopped one-by-one:
1. How does it detect new apps?
2. How does it handle duplicate connections as each app reports its sibling apps?
3. How does it handle apps being closed? Does user get any notification on app close?